src/main/java/uk/nhs/hee/tis/trainee/credentials/service/CachingDelegate.java

```diff
@@ -247,24 +247,25 @@ public Optional<String> getUnverifiedSessionIdentifier(UUID key) {
   }
 
   /**
-   * Cache a verified session identifier for later retrieval.
+   * Cache a verified session's identity unique identifier for later retrieval.
    *
-   * @param sessionId The identifier of the verified session to cache.
+   * @param sessionId  The identifier of the verified session to cache.
+   * @param identityId The verified identity's unique identifier.
    * @return The cached session identifier.
    */
-  @CachePut(cacheNames = VERIFIED_SESSION_IDENTIFIER, cacheManager = VERIFIED_SESSION_DATA)
-  public String cacheVerifiedSessionIdentifier(String sessionId) {
-    return sessionId;
+  @CachePut(cacheNames = VERIFIED_SESSION_IDENTIFIER, cacheManager = VERIFIED_SESSION_DATA, key = "#sessionId")
+  public UUID cacheVerifiedSessionIdentityIdentifier(String sessionId, UUID identityId) {
```

Comment:
Do we need to cache the `sessionId` in case a single user has more than one session open at a time? I was initially wondering if we needed it any more...